pkg/image/api/helper.go

```diff
@@ -191,6 +178,61 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 	return ref, nil
 }
 
+// ParseDockerImageReference parses a Docker pull spec string into a
+// DockerImageReference.
+func ParseDockerImageReference(spec string) (DockerImageReference, error) {
+	// TODO replace with docker version once docker/docker PR11109 is merged upstream
```

Comment:
Please document the special behavior in the godoc for these methods - anything this fundamental should be documented well.  Please include the official spec, and each point where we differ.